modules/module-mysql/test/src/MysqlTypeMappings.test.ts

```diff
@@ -0,0 +1,302 @@
+  const connectionManager = new MySQLConnectionManager(TEST_CONNECTION_OPTIONS, {});
+
+  afterAll(async () => {
+    await connectionManager.end();
+  });
+
+  async function setupTable() {
+    const connection = await connectionManager.getConnection();
+    await clearTestDb(connection);
+    await connection.query(`CREATE TABLE test_data (
+    tinyint_col TINYINT,
+    smallint_col SMALLINT,
+    mediumint_col MEDIUMINT,
+    int_col INT,
+    integer_col INTEGER,
+    bigint_col BIGINT,
+    float_col FLOAT,
+    double_col DOUBLE,
+    decimal_col DECIMAL(10,2),
+    numeric_col NUMERIC(10,2),
+    bit_col BIT(8),
+    boolean_col BOOLEAN,
+    serial_col SERIAL,
+    
+    date_col DATE,
+    datetime_col DATETIME(3),
+    timestamp_col TIMESTAMP(3),
+    time_col TIME,
+    year_col YEAR,
+
+    char_col CHAR(10),
+    varchar_col VARCHAR(255),
+    binary_col BINARY(16),
+    varbinary_col VARBINARY(256),
+    tinyblob_col TINYBLOB,
+    blob_col BLOB,
+    mediumblob_col MEDIUMBLOB,
+    longblob_col LONGBLOB,
+    tinytext_col TINYTEXT,
+    text_col TEXT,
+    mediumtext_col MEDIUMTEXT,
+    longtext_col LONGTEXT,
+    enum_col ENUM('value1', 'value2', 'value3'),
+    set_col SET('value1', 'value2', 'value3'),
+
+    json_col JSON,
+
+    geometry_col GEOMETRY,
+    point_col POINT,
+    linestring_col LINESTRING,
+    polygon_col POLYGON,
+    multipoint_col MULTIPOINT,
+    multilinestring_col MULTILINESTRING,
+    multipolygon_col MULTIPOLYGON,
+    geometrycollection_col GEOMETRYCOLLECTION
+    )`);
+
+    connection.release();
+  }
+
+  test('Number types mappings', async () => {
+    await setupTable();
+    await connectionManager.query(`
+INSERT INTO test_data (
+    tinyint_col,
+    smallint_col,
+    mediumint_col,
+    int_col,
+    integer_col,
+    bigint_col,
+ -- float_col,
+    double_col,
+    decimal_col,
+    numeric_col,
+    bit_col,
+    boolean_col
+    -- serial_col is auto-incremented and can be left out
+) VALUES (
+    127,                 -- TINYINT maximum value
+    32767,               -- SMALLINT maximum value
+    8388607,             -- MEDIUMINT maximum value
+    2147483647,          -- INT maximum value
+    2147483647,          -- INTEGER maximum value
+    9223372036854775807, -- BIGINT maximum value
+    -- 3.14,             -- FLOAT example
+    3.1415926535,        -- DOUBLE example
+    12345.67,            -- DECIMAL(10,2) example
+    12345.67,            -- NUMERIC(10,2) example
+    b'10101010',         -- BIT(8) example in binary notation
+    TRUE                 -- BOOLEAN value (alias for TINYINT(1))
+    -- serial_col is auto-incremented
+)`);
+
+    const databaseRows = await getDatabaseRows(connectionManager, 'test_data');
+    const replicatedRows = await getReplicatedRows();
+
+    const expectedResult = {
+      tinyint_col: 127,
+      smallint_col: 32767,
+      mediumint_col: 8388607,
+      int_col: 2147483647,
+      integer_col: 2147483647,
```

Comment:
For postgres (and mongodb, I think), we use bigint for all integer types. It would be best to use the same here.

That specifically helps to keep the distinction between the "integer" and "real" types in the SQLite type system, which we also use for sync rules.

Reply:
I see - so we just convert all integer number types to Bigint?

Reply:
Correct.